src/Symfony/Component/ExpressionLanguage/ParsedExpression.php

```diff
@@ -42,6 +42,17 @@ public function getNodes()
 
     public function dump()
     {
-        return $this->nodes->dump();
+        return $this->dumpNode($this->nodes);
+    }
+
+    protected function dumpNode(Node $node)
```

Comment:
the right extension point is probably to have this ``dumpNode`` method in a separate class, allowing to use a different implementation